### ISSUE TITLE
Drawing/Re-Drawing problem?

### ISSUE THREAD

**Issue by 3oheicrw** [snaildos/Fifo-Browser#111]
I don't know how it's called. I'm not a programmer. But if you set the Theme variant to Dark every actions in the address bar and tab bar area will have some "lags". For example, when search suggestions being shown, first they appeared as a white blank area before get populated with content. If you open more than one tabs and closed a tab, the entire address bar and tab bar area will be rendered white blanked a while before it switched you back to your closest tab. When watching youtube, if you change the CC to other language (default is English), it's also seems the whole window get repainted. The same lagging when you try to go full screen and back. It's somewhat laggy not as smooth as normal Chrome/Firefox. BTW, the video quality on youtube is good. I don't know anyway to check if hardware acceleration is working or not. Fifo doesn't support chrome://gpu. But it seems the youtube video has hardware acceleration but the Fifo graphical interface doesn't.

**Comment by 3oheicrw** [comment]
It's not always happening. Sometimes just resize the browser window triggered this. Browsing a particular site seems to trigger this more than other sites. I have no idea what is happening.

**Comment by snaildos** [comment]
This needs further infomation and I cannot reproduce.

**Comment by 3oheicrw** [comment]
> This needs further infomation and I cannot reproduce.

Yes. I think we should close this issue too. I'm also stuck. Can't give you anything more than that.

**Comment by 3oheicrw** [comment]
I reopened the issue. It seems it's really repainting/redrawing or something like that is the problem. New information: not only the Dark theme but the Light theme also suffers from this problem. It usually happens when switching between many tabs and you type a new address in one of the tab. The address bar area will be blank before it got updated. The whole page got blank when you switching between the tabs before they are rendered (some kind of repainting/redrawing is there). There are pages that easier to trigger the problem. These pages use some kind of JavaScript to update the page content dynamically.

**Comment by 3oheicrw** [comment]
Related issue: https://github.com/snaildos/Fifo-Browser/issues/113 https://github.com/snaildos/Fifo-Browser/issues/117

**Comment by 3oheicrw** [comment]
I noticed when I'm typing the reply in Github with Fifo I saw something wrong. The text "Attach files..." is animated somehow like it's constantly updating to match my typing. This is not happen with Min or Chrome/Firefox. There must be a problem but I don't have the knowledge to debug.

**Comment by 3oheicrw** [comment]
After using Min and Fifo in parallel for a few days I found Min is actually faster than Fifo. It could be because it's using a newer Electron (Chrome 102 vs Fifo is Chrome 95? I personally don't think Electron version matters), or it could be it was built natively on Windows with Visual Studio (very likely, but still not sure, I hear people said Chrome for Windows was built with Clang not Visual Studio and so does Firefox, someone even said modern Firefox for Windows was cross compiled from Linux, so cross compiling is currently the trend and I don't see my Firefox any slower than past versions).

**Comment by snaildos** [comment]
> After using Min and Fifo in parallel for a few days I found Min is actually faster than Fifo. It could be because it's using a newer Electron (Chrome 102 vs Fifo is Chrome 95? I personally don't think Electron version matters), or it could be it was built natively on Windows with Visual Studio (very likely, but still not sure, I hear people said Chrome for Windows was built with Clang not Visual Studio and so does Firefox, someone even said modern Firefox for Windows was cross compiled from Linux, so cross compiling is currently the trend and I don't see my Firefox any slower than past versions).

Min has no assets compared to Fifo's complex codespace. Fifo has tons of caching, bookmarks + adblcker and script detection tool. It could be the electron version, and I will be updating electron in v1.2.2 (Dependency Update) #116 
This hopefully fixes the issue.

**Comment by va0dl914** [comment]
> > After using Min and Fifo in parallel for a few days I found Min is actually faster than Fifo. It could be because it's using a newer Electron (Chrome 102 vs Fifo is Chrome 95? I personally don't think Electron version matters), or it could be it was built natively on Windows with Visual Studio (very likely, but still not sure, I hear people said Chrome for Windows was built with Clang not Visual Studio and so does Firefox, someone even said modern Firefox for Windows was cross compiled from Linux, so cross compiling is currently the trend and I don't see my Firefox any slower than past versions).
> 
> Min has no assets compared to Fifo's complex codespace. Fifo has tons of caching, bookmarks + adblcker and script detection tool. It could be the electron version, and I will be updating electron in v1.2.2 (Dependency Update) #116 This hopefully fixes the issue.

Min supports userscripts. You don't.
https://github.com/minbrowser/min/wiki/userscripts

**Comment by 3oheicrw** [comment]
> Min supports userscripts. You don't. https://github.com/minbrowser/min/wiki/userscripts

IMO userscripts is just a lightweight type of extensions. Still useful if we have it though. But the problem is reusing scripts for well known managers like greasemonkey, tampermonkey, violentmonkey,... Does Min's scripts manager compatible with them?

@snaildos What's your view about userscripts support?

**Comment by snaildos** [comment]
> > > After using Min and Fifo in parallel for a few days I found Min is actually faster than Fifo. It could be because it's using a newer Electron (Chrome 102 vs Fifo is Chrome 95? I personally don't think Electron version matters), or it could be it was built natively on Windows with Visual Studio (very likely, but still not sure, I hear people said Chrome for Windows was built with Clang not Visual Studio and so does Firefox, someone even said modern Firefox for Windows was cross compiled from Linux, so cross compiling is currently the trend and I don't see my Firefox any slower than past versions).
> > 
> > 
> > Min has no assets compared to Fifo's complex codespace. Fifo has tons of caching, bookmarks + adblcker and script detection tool. It could be the electron version, and I will be updating electron in v1.2.2 (Dependency Update) #116 This hopefully fixes the issue.
> 
> Min supports userscripts. You don't. https://github.com/minbrowser/min/wiki/userscripts

Min also has many contributors and developers. We don't. 🤦 😑 😮‍💨

**Comment by 3oheicrw** [comment]
> Min also has many contributors and developers. We don't. facepalm expressionless face_exhaling

Contrary to what you think. Submitted issues on Min's github repo rarely get the attention of the developer (PalmerAL) and I have closed a bunch of no reply issues submitted by me on their repo out of disappointment :smile:

**Comment by 3oheicrw** [comment]
> I mean, look. They are pretty cool, but then I will need to make a loader, official support, yadee yadee yaa, it's all annoying. This really is not a priority. Can we please discuss this in a new thread if you two believe userscripts are a "MUST HAVE" on Fifo?

No. I don't think userscripts is something "MUST HAVE". I only think they are good to have but not really needed. Don't confuse me with him. BTW, Fifo supports Chrome extension. Maybe he should try installing ViolentMonkey to see if it works :smile:

**Comment by snaildos** [comment]
This was an actual issue with Chrome, I had experienced it on my laptop, sites would get blurry for no reason, and after a minute or two, it would work. Since Fifo uses Chrome's rendering engine, Fifo is also affected. Give it a couple of months and Fifo+Chrome will both have a fix.

**Comment by snaildos** [comment]
> > > After using Min and Fifo in parallel for a few days I found Min is actually faster than Fifo. It could be because it's using a newer Electron (Chrome 102 vs Fifo is Chrome 95? I personally don't think Electron version matters), or it could be it was built natively on Windows with Visual Studio (very likely, but still not sure, I hear people said Chrome for Windows was built with Clang not Visual Studio and so does Firefox, someone even said modern Firefox for Windows was cross compiled from Linux, so cross compiling is currently the trend and I don't see my Firefox any slower than past versions).
> > 
> > 
> > Min has no assets compared to Fifo's complex codespace. Fifo has tons of caching, bookmarks + adblcker and script detection tool. It could be the electron version, and I will be updating electron in v1.2.2 (Dependency Update) #116 This hopefully fixes the issue.
> 
> Min supports userscripts. You don't. https://github.com/minbrowser/min/wiki/userscripts

Use MonkeyPatch or some other sort of Script Injector.
I don't see a point. 
With this attitude, I can just tell you: Fifo has extensions, Min doesn't. 🙄

**Comment by va0dl914** [comment]
> > Min supports userscripts. You don't. https://github.com/minbrowser/min/wiki/userscripts
> 
> Use MonkeyPatch or some other sort of Script Injector. I don't see a point. With this attitude, I can just tell you: Fifo has extensions, Min doesn't. roll_eyes

Min supports injection of arbitrary userscripts, you don't! :face_exhaling:

**Comment by va0dl914** [comment]
> This was an actual issue with Chrome, I had experienced it on my laptop, sites would get blurry for no reason, and after a minute or two, it would work. Since Fifo uses Chrome's rendering engine, Fifo is also affected. Give it a couple of months and Fifo+Chrome will both have a fix.

This is nonsense. I'm long time Chrome user and I have never met this issue. Blaming Chrome and Google doesn't make you feel better :-1:

**Comment by 3oheicrw** [comment]
> This was an actual issue with Chrome, I had experienced it on my laptop, sites would get blurry for no reason, and after a minute or two, it would work. Since Fifo uses Chrome's rendering engine, Fifo is also affected. Give it a couple of months and Fifo+Chrome will both have a fix.

I don't think so. I remember I got something like that with Chromium 94 (indeed it's a Chinese fork of Chromium called Advanced Chrome here: https://browser.taokaizen.com/ but it's somewhat never stable IMO, the sound is louder than normal Chrome/Chromium so I used it only to watch youtube). The problem is apart from this particular Chromium fork, I have never experienced anything similar with Google Chrome nor other Chromiums on this system.

**Comment by snaildos** [comment]
> > This was an actual issue with Chrome, I had experienced it on my laptop, sites would get blurry for no reason, and after a minute or two, it would work. Since Fifo uses Chrome's rendering engine, Fifo is also affected. Give it a couple of months and Fifo+Chrome will both have a fix.
> 
> I don't think so. I remember I got something like that with Chromium 94 (indeed it's a Chinese fork of Chromium called Advanced Chrome here: https://browser.taokaizen.com/ but it's somewhat never stable IMO, the sound is louder than normal Chrome/Chromium so I used it only to watch youtube). The problem is apart from this particular Chromium fork, I have never experienced anything similar with Google Chrome nor other Chromiums on this system.

No, it 100% is. Try the new update, pretty sure that will fix it.
https://support.google.com/chrome/answer/95290?hl=en
It's a known thing with Chrome, no clue if its fixed or not.

**Comment by snaildos** [comment]
> > This was an actual issue with Chrome, I had experienced it on my laptop, sites would get blurry for no reason, and after a minute or two, it would work. Since Fifo uses Chrome's rendering engine, Fifo is also affected. Give it a couple of months and Fifo+Chrome will both have a fix.
> 
> This is nonsense. I'm long time Chrome user and I have never met this issue. Blaming Chrome and Google doesn't make you feel better 👎

How is it nonsense? 
I have 100% had it on my laptop, it has happened before. The only thing nonsense is your attitude towards me. 
It's honestly disgusting the way you have acted towards me in this issues. If you want to talk to me like this, just honestly don't.
I spend hours upon hours on this project with nothing in return, I don't need more downsides to making this browser. It makes me just want to abandon the project. What is your problem? I don't see why you are acting like this, not only in this issue, but all issues you have chatted in. 🤦

**Comment by snaildos** [comment]
> > > Min supports userscripts. You don't. https://github.com/minbrowser/min/wiki/userscripts
> > 
> > 
> > Use MonkeyPatch or some other sort of Script Injector. I don't see a point. With this attitude, I can just tell you: Fifo has extensions, Min doesn't. roll_eyes
> 
> Min supports injection of arbitrary userscripts, you don't! 😮‍💨

Code it yourself 🤣 😆 
I said I already discussed the status on that idea.
I have decided I won't be adding scripting into Fifo!
I'll accept a PR for it though.
